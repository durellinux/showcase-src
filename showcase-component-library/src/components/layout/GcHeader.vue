// Page header component.
// It is used in the layout to display the page title and the page actions.
// Page title and page actions are passed as slots.
// The background is set to the primary color as defined in the design system.
// The title is aligned to the left, while the actions are aligned to the right.
// The minimum height of the header is 100px.

<template>
  <header class="header">
    <div class="header__title">
      <slot name="title" />
    </div>
    <div class="header__actions">
      <slot name="actions" />
    </div>
  </header>
</template>

<script lang="ts" setup>
</script>

<style lang="scss" scoped>
@import "../design_system_tokens.scss";

.header {
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 0 1rem;
  background-color: $color-primary-darkest;
  color: $color-text-light;
  height: 100px;
}

.header__title {
  @include typography-heading-3;
}

.header__actions {
  display: flex;
  align-items: center;
  gap: 1rem;
}
</style>
