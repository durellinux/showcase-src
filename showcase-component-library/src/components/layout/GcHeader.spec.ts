// Tests for GcHeader component

import { mount } from '@vue/test-utils';
import GcHeader from './GcHeader.vue';

describe('GcHeader', () => {
    test('Renders title slot', () => {
        const wrapper = mount(GcHeader, {
            slots: {
                title: 'Title',
            },
        });
        expect(wrapper.text()).toContain('Title');
    });

    test('Renders Actions slot', () => {
        const wrapper = mount(GcHeader, {
            slots: {
                actions: 'Actions',
            },
        });
        expect(wrapper.text()).toContain('Actions');
    });

    test('Renders complex title slot', () => {
        const wrapper = mount(GcHeader, {
            slots: {
                title: '<span>Complex</span> <span>title</span>',
            },
        });
        expect(wrapper.text()).toContain('Complex title');
    });

    test('Renders complex Actions slot', () => {
        const wrapper = mount(GcHeader, {
            slots: {
                actions: '<span>Complex</span> <span>actions</span>',
            },
        });
        expect(wrapper.text()).toContain('Complex actions');
    });

    test('Matches snapshot', () => {
        const wrapper = mount(GcHeader, {
            slots: {
                title: 'Title',
                actions: 'Actions',
            },
        });
        expect(wrapper.element).toMatchSnapshot();
    });
});
