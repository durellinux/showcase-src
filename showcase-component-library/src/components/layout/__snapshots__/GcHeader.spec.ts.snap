// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GcHeader Matches snapshot 1`] = `
<header
  class="header"
>
  <div
    class="header__title"
  >
    
    Title
    
  </div>
  <div
    class="header__actions"
  >
    
    Actions
    
  </div>
</header>
`;
